### PR TITLE
improvement: add better numbering support for snakecasing when smartCasing is enabled

### PR DESCRIPTION
**Pull request by armandobelardo**
A couple examples are in the comment, but currently we use lodash snakeCase, which turns "v2" into "v_2", this adds a small function for the `smartCasing` flag that allows "v2" to stay "v2", and a couple other examples: "test2This2 2v22" => "test2this2_2v22", "applicationV1" => "application_v1"

### CODE REVIEW

File: packages/cli/generation/ir-generator/src/casings/CasingsGenerator.ts [hunk 1]
```diff
@@ -24,38 +24,48 @@ export function constructCasingsGenerator({
...
+                        .join("");
+                }
+
+                // In smartCasing, manage numbers next to letters differently:
+                // _.snakeCase("v2") = "v_2"
+                // smartCasing("v2") = "v2", other examples: "test2This2 2v22" => "test2this2_2v22", "applicationV1" => "application_v1"
+                snakeCaseName = name
+                    .split(" ")
+                    .map((part) => part.split(/(\d+)/).map(snakeCase).join(""))
+                    .join("_");
```
Comment (dsinghvi) [inline packages/cli/generation/ir-generator/src/casings/CasingsGenerator.ts:68]: @armandobelardo can we add some tests here ?

Comment (dsinghvi) [reply packages/cli/generation/ir-generator/src/casings/CasingsGenerator.ts:68]: https://github.com/fern-api/fern/blob/main/packages/cli/generation/ir-generator/src/__test__/casingsGenerator.test.ts

**Comment by armandobelardo** [comment]
@dsinghvi test added

**Review by dsinghvi — APPROVED** [review]

**Comment by dsinghvi** [comment]
@armandobelardo looks like you need to update some snapshots `yarn workspace @fern-api/ir-generation test -u`

**Pull request merged**